sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -0,0 +1,174 @@
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"aws", "gcp", "azure"}, false),
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func getSecureCloudauthAccountClient(client SysdigClients) (v2.CloudauthAccountSecureInterface, error) {
+	return client.sysdigSecureClientV2()
+}
+
+func resourceSysdigSecureCloudauthAccountCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureCloudauthAccountClient((meta.(SysdigClients)))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudauthAccount, err := client.CreateCloudauthAccountSecure(ctx, cloudauthAccountFromResourceData(data))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(cloudauthAccount.Id)
+
+	return nil
+}
+
+func resourceSysdigSecureCloudauthAccountRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureCloudauthAccountClient(meta.(SysdigClients))
+
+	if err != nil {
+		data.SetId("")
+		return diag.FromErr(err)
+	}
+
+	cloudauthAccount, err := client.GetCloudauthAccountSecure(ctx, data.Id())
+
+	if err != nil {
+		data.SetId("")
+
+		if strings.Contains(err.Error(), "404") {
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	err = cloudauthAccountToResourceData(data, cloudauthAccount)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceSysdigSecureCloudauthAccountUpdate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureCloudauthAccountClient(meta.(SysdigClients))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = client.UpdateCloudauthAccountSecure(ctx, data.Id(), cloudauthAccountFromResourceData(data))
+
+	if err != nil {
+		if strings.Contains(err.Error(), "404") {
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceSysdigSecureCloudauthAccountDelete(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureCloudauthAccountClient(meta.(SysdigClients))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = client.DeleteCloudauthAccountSecure(ctx, data.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func cloudauthAccountFromResourceData(data *schema.ResourceData) *v2.CloudauthAccountSecure {
+	return &v2.CloudauthAccountSecure{
+		Id:         data.Get("account_id").(string),
```

Comment:
It's fetched from the resourceData but we're using this function to pass the cloud account object for account creation.

Reply:
Yes, I think the `id` field will be computed from the BE http API and returned back.

- We won't need it here in the `From` func, i.e we won't need this input from the terraform resourceData.
- Adding it in `To` func in L146 looks correct. Since there we are populating in the return path, what we get back as response from the BE http API.

Also, shall we call the resource field name as `account_id` or just `id` in the schema to avoid any confusion? thoughts?

Reply:
let's keep it as what is defined in proto i.e. Id

Reply:
The proto differentiates the IDs as `id`, `provider_id` and so on. In that order, I'd prefer having it as `account_id` as it's fairly descriptive and easily distinguishable from `organization_id` or any other ID fields.